daemon/api_access_control.go

```diff
@@ -63,28 +63,41 @@ func getRules(c *Command, r *http.Request, user *auth.UserState) Response {
 	}
 
 	query := r.URL.Query()
+
+	snap := query.Get("snap")
+	app := query.Get("app")
+
 	follow := false
 	if s := query.Get("follow"); s != "" {
 		f, err := strconv.ParseBool(s)
 		if err != nil {
 			return BadRequest("invalid value for follow: %q: %v", s, err)
 		}
+		if snap == "" {
+			return BadRequest("follow=true parameter provided, must also provide snap parameter")
```

Comment:
Is following of all snaps going to be supported in the future?

Reply:
It very well could be. There were worries about exceeding the size of a single message by sending the backlog of rules all at once, but the current implementation of json-seq in this PR sends existing rules one at a time, as if they were new rules being added, so this should not be a problem.

I'll rework the `FollowRulesSeqResponseWriter` to allow this (should be relatively simple) but I'll leave the spec as-is for now, and add this as a question to discuss in the future. Thanks!

Reply:
Further info on why it would be useful to follow all at once: In gnome-control-center we want to know when rules are added/removed to update the UI. If we have to do it per-snap then we need multiple connections to snapd to do this. However, in the current design we only show one app at once, so this is not an immediate concern. This assumption could change in the future though.